add-ons/istio/istio.addon

```diff
@@ -1,37 +1,31 @@
 # Name: istio
 # Description: Deploys Latest Istio
 # Url: https://github.com/istio/istio
-# Required-Vars: ISTIO_HOME
-# OpenShift-Version: >=3.7.0
-
+# OpenShift-Version: >=3.10.0
+# Depends-On: dynamic-admission-controllers
 # From istio doc
-oc adm policy add-scc-to-user anyuid -z istio-ingress-service-account -n istio-system
-oc adm policy add-scc-to-user anyuid -z istio-egress-service-account -n istio-system
-oc adm policy add-scc-to-user anyuid -z istio-pilot-service-account -n istio-system
-oc adm policy add-scc-to-user anyuid -z default -n istio-system
 
-# Create Istio 
-oc create -f #{ISTIO_HOME}/install/kubernetes/istio.yaml
+echo Prepare for install istio
+
+ssh sudo sysctl vm.max_map_count=262144
 
-oc expose svc istio-ingress -n istio-system
+echo Install istio-operator
 
-# Add Istio Addons 
-oc create -f #{ISTIO_HOME}/install/kubernetes/addons/prometheus.yaml
-oc create -f #{ISTIO_HOME}/install/kubernetes/addons/grafana.yaml
-oc create -f #{ISTIO_HOME}/install/kubernetes/addons/servicegraph.yaml
-oc expose svc grafana -n istio-system
-oc expose svc servicegraph -n istio-system
-oc expose svc prometheus -n istio-system
+!oc new-project istio-operator
+!oc new-app -f istio_community_operator_template.yaml -l 'minishift-addon=istio' --param=OPENSHIFT_ISTIO_MASTER_PUBLIC_URL=https://#{ip}:8443/
 
-# Install Jaeger
-oc apply -f templates/jaeger-all-in-one.yml -n openshift
-oc new-app jaeger-template-all-in-one -n istio-system
+sleep 10
 
-# Add the any uid permission to the default myproject
-oc adm policy add-scc-to-user anyuid -z default -n myproject
-oc adm policy add-scc-to-user privileged -z default -n myproject
+echo Install istio
+
+oc apply -f installation.yaml
 
 oc project myproject
 
-echo 'Please wait for few seconds before all pods are up!'
-echo 'Watch the pods status via minishift console or oc get pods -w --all-namespaces --as system:admin'
\ No newline at end of file
+echo Please wait for few seconds before all pods are up!
+echo Watch the pods status via minishift console or oc get pods -w -n istio-system --as system:admin
+
+echo Please update the Security Context Constraints (SCC) for using istio side-car in your project.
+echo oc adm policy add-scc-to-user anyuid -z default -n myproject
+echo oc adm policy add-scc-to-user privileged -z default -n myproject
+echo oc label namespace myproject istio-injection=enabled
```

Comment:
i guess the istio-injection enabled settings here is kind of mixed up with Knative in mind, I would feel these should be moved out add-on installation, as that really depends on where the user wants to deploy the applications. Put in the README as an example rather than add-on doing it automatically for the user.